layers.go

```diff
@@ -176,7 +176,7 @@ func (bp *bpLayer) hasLocalContents() bool {
 }
 
 func (bp *bpLayer) writeSha(sha string) error {
-	if err := ioutil.WriteFile(bp.path+".sha", []byte(sha), 0777); err != nil {
+	if err := ioutil.WriteFile(bp.path+".sha", []byte(sha), 0777); err != nil { //nolint:gosec
```

Comment:
I am pretty sure we still need `0777` as we can't be sure of the user who is going to read this file - but I am happy to be corrected.

Reply:
I think we could get away with `666` - `777` is only needed if we are going to need execute permissions on the `.sha` file, which I don't believe we need.

Reply:
Haha `0666` still requires `//nolint` but that's a good point! I think we could make this change as it's a step in the right direction.